src/release/breaking-changes/flutter-gradle-plugin-apply.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Deprecated imperative apply of Flutter Gradle Plugin
+description: >
+  How to migrate your Android app's Gradle build files to the new, declarative
+  format.
+---
+
+## Summary
+
+To build a Flutter app for Android, the Flutter Gradle plugin must be applied.
+Historically, this was done imperatively with Gradle's
+[legacy, imperative apply method][].
+
+In Flutter 3.16, support has been added for applying these plugins with Gradle's
+[declarative plugins {} block][], and it is the recommended approach. Since
+Flutter 3.16, projects generated with `flutter create` use the new apply method
+of applying Gradle plugins. Older projects, however, should be migrated
+manually.
+
+To learn about advantages the new declarative apply syntax has over the old
+imperative apply syntax, see [Gradle docs about plugins {} block][plugins block].
+
+Migrating the app ecosystem to use the new approach will also make it easier for
+Flutter team to develop Flutter's Gradle plugins, and to enable exciting new
+features in the future, such as using Kotlin instead of Groovy in Gradle
+buildscripts.
+
+## Migrate
+
+### android/settings.gradle
+
+Replace contents with:
+
+```gradle
+pluginManagement {
+    def flutterSdkPath = {
+        def properties = new Properties()
+        file("local.properties").withInputStream { properties.load(it) }
+        def flutterSdkPath = properties.getProperty("flutter.sdk")
+        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
+        return flutterSdkPath
+    }
+    settings.ext.flutterSdkPath = flutterSdkPath()
+
+    includeBuild("${settings.ext.flutterSdkPath}/packages/flutter_tools/gradle")
+
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
+
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "7.3.0" apply false
```

Comment:
Sure – would this:
```gradle
id "com.android.application" version "{agpVersion}" apply false
id "org.jetbrains.kotlin.android" version "{kotlinVersion}" apply false
```
be ok?

PS I was also thinking about including the newest possible versions of AGP and Kotlin here, because I think many people are not aware these things actually get updates and are stuck on AGP v4, haha.

Reply:
Only a quick thought, don't know if it's practical and can be ignored: can't we make a test, if the docs code is equivalent to the template, so it would replicate the up to date version of this one. But it does of course not work with more complicated code snippets like below.

I always find it helpful to know what is the Flutter supported version or at least, have a link to the updated template file.

Reply:
That'd be great – but I think it's out of scope of this PR (it deserves a separate issue imho)

Reply:
Please remove this version before merging. 
"{{agpVersion}} works for me so does <AGP-Version> to mirror how you document <app-src>

Reply:
done in 35e58151a062f23a009220fab778b99998d3dadc